athena-jdbc/src/main/java/com/amazonaws/athena/connectors/jdbc/connection/DatabaseConnectionConfigBuilder.java

```diff
@@ -123,13 +131,12 @@ private DatabaseConnectionConfig extractDatabaseConnectionConfig(final String ca
 
         Validate.notBlank(dbType, "Database type must not be blank.");
         Validate.notBlank(jdbcConnectionString, "JDBC Connection string must not be blank.");
-
-        JdbcConnectionFactory.DatabaseEngine databaseEngine = JdbcConnectionFactory.DatabaseEngine.valueOf(dbType.toUpperCase());
+        Validate.isTrue(dbType.equals(this.engine), "JDBC Connection string must be prepended by correct database type.");
```

Comment:
I might have missed it, but is there a new unit test(s) that verifies this exception when the wrong type is provided?

Reply:
There is one [here](https://github.com/awslabs/aws-athena-query-federation/pull/662/files#diff-9bc3afe8e5d57274dbbbe995b87caee973e97d142a71f59082d9dabed72b4773R38)